constraint-solver/src/algebraic_constraint/solve.rs

```diff
@@ -159,27 +154,27 @@ where
         &self,
         range_constraints: &impl RangeConstraintProvider<T::FieldType, V>,
     ) -> Result<ProcessResult<T, V>, Error> {
-        let expression = self.expression;
+        let (_, mut linear, constant) = self.expression.components();
```

Comment:
Ah right, here it is. A struct would be nice.

Reply:
`components()` has been used quite a lot in the code, but yes, probably better to have a struct for that.